api/src/main/java/org/openmrs/validator/TagValidator.java

```diff
@@ -0,0 +1,56 @@
+package org.openmrs.validator;
+
+import org.openmrs.Tag;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.TagService;
+import org.openmrs.api.context.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import static org.apache.solr.core.SolrConfig.log;
+
+@Handler(supports = { Tag.class })
+public class TagValidator implements Validator {
+	
+	@Override
+	public boolean supports(Class<?> aClass) {
+		return Tag.class.isAssignableFrom(aClass);
+	}
+	
+	/**
+	 * checks if a given Tag object is valid
+	 * 
+	 * @see org.springframework.validation.Validator#validate(java.lang.Object,
+	 *      org.springframework.validation.Errors)
+	 * @should fail if tag name is empty
+	 * @should fail if object_uuid is empty
+	 * @should fail if object_type is empty
+	 * @should fail if object_type with object_uuid does not exist in database
+	 */
+	@Override
+	public void validate(Object obj, Errors errors) {
+		if (obj == null || !(obj instanceof Tag)) {
+			throw new IllegalArgumentException("The parameter obj should not be null and must be of type" + Tag.class);
+		}
+		Tag tagInstance = (Tag) obj;
+		
+		ValidationUtils.rejectIfEmpty(errors, "tag", "Tag.tag.empty");
+		ValidationUtils.rejectIfEmpty(errors, "objectUuid", "Tag.object_uuid.empty");
+		ValidationUtils.rejectIfEmpty(errors, "objectType", "Tag.object_type.empty");
+		
+		try {
+			if (!Context
```

Comment:
Try to clean up this do that you ave shorter lines, get the service and have in a local variable and then call that